recipe/meta.yaml

```diff
@@ -26,10 +26,32 @@ requirements:
     - cmake
     - ninja
 
+{% set absl_libs = [
+    "base", "civil_time", "cord", "cordz_functions", "cordz_handle", "cordz_info",
+    "cordz_sample_token", "examine_stack", "exponential_biased", "failure_signal_handler",
+    "flags", "flags_commandlineflag", "flags_config", "flags_marshalling", "flags_parse",
+    "flags_private_handle_accessor", "flags_program_name", "flags_reflection", "flags_usage",
+    "hash", "hashtablez_sampler", "int128", "log_severity", "low_level_hash", "periodic_sampler",
+    "random_distributions", "random_seed_gen_exception", "random_seed_sequences", "raw_hash_set",
+    "scoped_set_env", "spinlock_wait", "stacktrace", "status", "statusor", "strerror", "strings",
+    "symbolize", "synchronization", "time", "time_zone"
+] %}
 test:
   commands:
-    - test -f $PREFIX/lib/libabsl_base${SHLIB_EXT}  # [not win]
-    - if not exist %LIBRARY_LIB%\\absl_base.lib exit 1  # [win]
+    # windows
+    - if not exist %LIBRARY_LIB%\\abseil_dll.lib exit 1  # [win]
+    - if not exist %LIBRARY_BIN%\\abseil_dll.dll exit 1  # [win]
+
+    # absl_libs
+    {% for each_lib in absl_libs %}
+    # presence of shared libs
+    - test -f $PREFIX/lib/libabsl_{{ each_lib }}${SHLIB_EXT}      # [unix]
+    - if not exist %LIBRARY_LIB%\\absl_{{ each_lib }}.lib exit 1  # [win]
+    # absence of static libs
+    - test ! -f $PREFIX/lib/libabsl_{{ each_lib }}.a              # [unix]
+    # TODO: do we want to get rid of the static windows libs?
+    # - if exist %LIBRARY_LIB%\\absl_{{ each_lib }}.lib exit 1      # [win]
```

Comment:
On Windows, you will always have `.lib` files. In the case of shared DLLs, these are small import libraries needed that only contain "glue" to lazy load the function definitions from the DLLs at runtime.

Reply:
I had to learn already that `.lib` is both for static libs as well as the "import libraries" that wrap DLLs, but the logs here look pretty unambiguous to me:
```
[142/154] Linking CXX static library absl\flags\absl_flags_parse.lib
```

Reply:
It seems there's only one DLL (plus its import lib) being created on windows with `BUILD_SHARED_LIBS`

Reply:
It's been some time since I last looked at the abseil build system, but my recollection was that on Windows only static builds were supported. Has this changed in the last year or so?

Reply:
Seems so. `abseil_dll.dll` is pretty unambiguous as well IMO. ;-)

Reply:
I am probably being paranoid, and I understand that this would be additional work for you, but would it be possible to have a small tests that checks (on Windows) that the DLL actually works (i.e., compile and run a small executable which calls functions that have been compiled in the DLL)?

I am asking because I am a bit scared by some of the statements in the abseil docs regarding dynamics libraries:

https://abseil.io/about/compatibility

But in any case, thanks a lot for working on this! Dynamic libraries are definitely the way to go for a distribution like conda, it would be great to have this PR merged.

Reply:
I would be personally more in favour of pulling the trigger and merge this PR directly, and perhaps roll it back later in case it turns out it breaks existing packages.

But certainly @xhochy has a better feeling than I do about whether this course of action is appropriate for conda-forge as a whole, so I'll leave the decision to him.